### ISSUE TITLE
Execution fails, but exit code is zero

### ISSUE THREAD

**Issue by acecilia** [SwiftGen/SwiftGen#911]
Hi 👋 

#### Expectation:
Running swiftgen will return error code 1 if any errors are found

#### Reality:
I observed that running swiftgen with a configuration file that misses some inputs results in an error log, but command returns zero:
```
$ swiftgen
swiftgen: error: File .../en.lproj/Localizable.strings not found.
swiftgen: error: It seems like there was an error running SwiftGen.

- Verify that your configuration file exists at the correct path, or create a new one using:
> swiftgen config init

- Verify that your configuration file is valid by running:
> swiftgen config lint

- If you have any other questions or issues, we have extensive documentation and an issue tracker on GitHub:
> https://github.com/SwiftGen/SwiftGen
$ echo $?                                                                                   
0
```

Swiftgen version: `6.5.1`

**Comment by djbe** [comment]
Good catch. I'm not 100% sure how to rework our code 🤔 

- We run all the sub-commands in parallel and collect the errors ✅ 
- After the execution, if we have any error(s), we throw it ✅ 
- The (config) run command catches this error and prints it nicely ✅ 
- It also prints the help message you see above ✅ (link https://github.com/SwiftGen/SwiftGen/blob/stable/Sources/SwiftGen/Commander/ConfigCommands.swift#L58-L61)
- We don't rethrow the caught error ❌ 

That last bit doesn't let `Commander` (framework) know something happened, so it'll `exit(0)`.

----

The problem is, if we were to simply rethrow our error, `Commander` will ALSO print it, so we'd get this as output:
> swiftgen: error: File .../en.lproj/Localizable.strings not found.
> ... help message ...
> An error occurred: File .../en.lproj/Localizable.strings not found.

Far from ideal of course. Moving the error to after the help message would be weird for the user. I can see a few options:
- `exit(1)` from inside that `catch`, will have to do this in other places
- Throw a generic simple error, so at the end we'd get "An error occurred: generic simple error goes here"

**Comment by djbe** [comment]
Implemented by SwiftGen 6.6.0, which has been released.
